pkg/event/registry.go

```diff
@@ -0,0 +1,142 @@
+package event
+
+import (
+	"github.com/pkg/errors"
+
+	k8sV1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/admiral/pkg/log"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+)
+
+type Registry struct {
+	name          string
+	networkPlugin string
+	eventHandlers []Handler
+}
+
+// NewRegistry creates a new registry, the name would normally be the name of the owner, for example
+// 'routeagent', and the networkPlugin must be the network plugin that has been detected (or passed
+// via env-var) this will be used to only keep the relevant handlers in runtime, while discarding
+// the ones that shall not be ran with the system network plugin.
+func NewRegistry(name, networkPlugin string) Registry {
+	return Registry{
+		name:          name,
+		networkPlugin: networkPlugin,
+		eventHandlers: []Handler{},
+	}
+}
+
+func (er *Registry) addHandler(eventHandler Handler) {
+	evNetworkPlugin := eventHandler.GetNetworkPlugin()
+
+	if evNetworkPlugin == AnyNetworkPlugin || evNetworkPlugin == er.networkPlugin {
+		er.eventHandlers = append(er.eventHandlers, eventHandler)
+		klog.Infof("Event handler %q loaded for the %q registry ",
+			eventHandler.GetName(), er.name)
+
+		if err := eventHandler.Init(); err != nil {
+			klog.Fatalf("Event handler %q failed to initialize: %v", eventHandler.GetName(), err)
+		}
+	} else {
+		klog.V(log.DEBUG).Infof("Event handler %q ignored for the %q registry ",
+			eventHandler.GetName(), er.name)
+	}
+}
+
+// AddHandlers adds multiple event.Handlers to the registry, the event handlers that don't correspond to
+// the runtime network plugin will be ignored, and the ones that work with the runtime network plugin
+// will be stored on the registry, so later method invocations will be dispatched to the handler.
+func (er *Registry) AddHandlers(eventHandlers ...Handler) {
+	for _, eventHandler := range eventHandlers {
+		er.addHandler(eventHandler)
+	}
+}
+
+func (er *Registry) StopHandlers(uninstall bool) error {
+	return er.invokeHandlers("Stop", func(h Handler) error {
```

Comment:
Instead of event names (aka Stop), it will be better to use constants/macros here as commented above.

Reply:
I prefer the practice of separating both. In this case I believe we don't really need the constants, because those are not used anywhere else (except for testing...), so we'd only be adding lines. 

Even on the testing side may be it's worth removing the constants into literals to save lines

Reply:
No, in testing we can't remove the constants we use them at least twice each.